### DISCUSSION TITLE
List of dirty properties

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3048] [Questions]
Hi,
Is there a way to get a list of dirty properties in an object?
In our use case, a data entry person updates a record and the record goes for approval to a manager.
The manager only wants to see the updated fields(properties) in the UI.

We have implemented @JasonBock 's solution of dirty properties in our project.
[http://www.jasonbock.net/News/Item/9cc70d85-bef3-4e2b-9a68-3ba82615f8a3](url)

Kind regards

**Comment by rockfordlhotka** [comment]
I'm at a conference right now, so don't have time to look through the code - but it seems to me that inside a domain object you have access to the object's property or field manager as a protected property. That should give you access to the list of field data instances that manage the field/property data for the domain object, and I think that'll get you to whatever info you have in those object - like your per-field isdirty indicator.

**Reply by Chicagoan2016** [reply]
Thank you @rockfordlhotka , as far as I understand : ), FieldManager gives us GetRegisteredProperties() and we could iterate the list of properties and call GetFieldData()
but I wonder if I am overthinking it?

Kind regards

**Reply by rockfordlhotka** [reply]
I think so. Though there is a [IsFieldDirty](https://github.com/MarimerLLC/csla/blob/825690d952f2212e25cbf244960f142b9994ad1a/Source/Csla/Core/FieldManager/FieldDataManager.cs#L416) method on `FieldManager` that might be useful?
